src/options.rs

```diff
@@ -0,0 +1,284 @@
+use syn;
+use quote;
+
+#[derive(PartialEq, Debug)]
+pub enum SetterPattern {
+    Owned,
+    Mutable,
+    Immutable
+}
+
+impl Default for SetterPattern {
+    fn default() -> SetterPattern {
+        SetterPattern::Mutable
+    }
+}
+
+#[derive(Debug)]
+pub struct Options {
+    /// e.g. `#[setters]` (defaults to true)
+    setter_enabled: bool,
+    /// e.g. `#[setters(owned)]` (defaults to mutable)
+    setter_pattern: SetterPattern,
+    /// e.g. `#[setters(prefix="with")]` (defaults to None)
+    setter_prefix: Option<String>,
+    /// e.g. `#[setters(private)]` (defaults to public)
+    setter_public: bool,
+    /// e.g. `#[setters(options="implicit")]` (defaults to explicit)
+    setter_implicit_options: bool,
+    /// e.g. `#[getters]` (defaults to false)
+    getter_enabled: bool,
+    /// e.g. `#[getters(prefix="with")]` (defaults to None)
+    getter_prefix: Option<String>,
+    /// e.g. `#[getters(private)]` (defaults to public)
+    getter_public: bool,
+}
+
+impl Options {
+    pub fn setter_pattern(&self) -> &SetterPattern {
+        &self.setter_pattern
+    }
+
+    pub fn setter_visibility(&self) -> Option<quote::Tokens> {
+        if self.setter_public {
+            Some(quote!(pub))
+        } else {
+            None
+        }
+    }
+
+    pub fn setter_enabled(&self) -> bool {
+        self.setter_enabled
+    }
+}
+
+impl<T> From<T> for Options where
+    T: IntoIterator<Item=syn::Attribute>
+{
+    fn from(attributes: T) -> Self {
+        trace!("Parsing struct attributes.");
+        let mut ob = OptionsBuilder::default();
+
+        for attr in attributes {
+            ob.parse_attribute(attr);
+        }
+
+        ob.into()
+    }
+}
+
+#[derive(Default, Debug)]
+pub struct OptionsBuilder {
```

Comment:
Heh, this code looks like a great indicator for a new "indirect builder" setting!

(Indirect builders are the "original" builder pattern using a struct with the same fields but all are optional.)